### DISCUSSION TITLE
I'd like to update form.formState with a custom function. Can you help please?

### DISCUSSION THREAD

**Discussion by npearson72** [react-hook-form/react-hook-form#9802] [Q&A] [answered]
I want to add a `isLoading` function on an instance of `form.formState` from `useForm`, like so:

```ts
  form.formState.isLoading = () => {
    return form.formState.isSubmitting || isLoading; // The `isLoading` state comes from React Query
  };
```

But I'm getting a typescript error:

<img width="706" alt="Screen Shot 2023-01-19 at 1 53 49 AM" src="https://user-images.githubusercontent.com/1323414/213386086-3d7a7cc1-84ab-4bae-be0d-7edc880f25b2.png">

Is there a way to handle this? Or perhaps... is there a better approach to this problem? My form is spread across multiple components in an accordion, so I'm trying to avoid prop drilling.

**Comment by bluebill1049** [comment, marked as answer]
maybe it's easier just to host your own state.

**Reply by npearson72** [reply]
I realized what I could do is return a promise from my onSubmit handler and that makes the `formState.isSubmitting` work until the promise is resolved.

Good thinking on your part to account for that! Cheers.
